src/servers/Http1Server.cc

```diff
@@ -120,8 +120,9 @@ Http::One::Server::buildHttpRequest(Http::StreamPointer &context)
             // else use default ERR_INVALID_REQ set above.
             break;
         }
-        // setLogUri should called before repContext->setReplyToError
-        setLogUri(http, http->uri, true);
+
+        assert(http->log_uri && http->al->effectiveVirginUrl()); // already initialized via ConnStateData::abortRequestParsing()
```

Comment:
Please split the a-and-b assert into two so that we know which condition has failed.

Please restore the `setReplyToError() requires log_uri` comment for the `log_uri` assert to explain why we are asserting that condition.

If the second (effectiveVirginUrl) condition is not really necessary for this code, then we may want to drop it. Otherwise, it would be nice to explain why we are willing to kill Squid for violating that condition.